modules/gltf/gltf_document.cpp

```diff
@@ -452,13 +454,14 @@ Error GLTFDocument::_serialize_nodes(Ref<GLTFState> state) {
 }
 
 String GLTFDocument::_sanitize_scene_name(Ref<GLTFState> state, const String &p_name) {
+#ifdef MODULE_REGEX_ENABLED
```

Comment:
I think if regex is disabled, it should still check `state->use_legacy_names`, and print a warning before falling back to the new name validation.

Here it will silently ignore what users configured themselves, breaking their imported files without notice.